### PR TITLE
BelaScope: add documentation

### PR DESCRIPTION
**Pull request by elgiano**
Purpose and Motivation
----------------------
Adding documentation for BelaScope

Types of changes
----------------

- Documentation (non-code change which corrects or adds documentation for existing features)

Checklist
---------

- [x] All tests are passing
- [x] If necessary, new tests were created to address changes in PR, and tests are passing
- [x] Updated documentation, if necessary
- [x] This PR is ready for review

Remaining Work
--------------

### CODE REVIEW

**Comment by giuliomoro** [comment]
Thanks.

A style note: you'll want to correct the case of the `Bela` word: you use `BeLa` or `BeLA`, but it's actually `Bela`.

As you know, I don't know much about Sc or its documentation style, so I am a bit surprised it needs adding docs in so many different files. I am wondering if this documentation is going to show up even to people that are not running on Bela (e.g.: within the Sc IDE on a Mac), in which case I see how this could raise some concerns when trying to mainline it.
@brianlheim what do you think?

**Comment by mossheim** [comment]
@giuliomoro this is fine as-is. i agree it could be confusing, but we already do this in mainline, so if it's a problem to be solved, it can't be solved here.

File: HelpSource/Classes/Array.schelp [hunk 1]
```diff
@@ -404,3 +404,13 @@ code::
...
+subsection:: BeLa
+
+method:: belaScope
+Send this Array's content to BeLa's Oscilloscope (see link::Classes/BelaScope:: for required setup)
+argument:: scopeChannel
+Bela's oscilloscope channel to start scoping on.
+
+argument:: server
+The server on which BelaScope is running. If not specified, it looks for the first server for which BelaScope was already initialized. If none is found, it attempts to initialize a link::Classes/BelaScope:: instance on link::Server#*default::.
+returns:: This Array.
```
Comment (mossheim) [inline HelpSource/Classes/Array.schelp]: make sure all your files have an ending newline

**Comment by elgiano** [comment]
Oh yeah, I so much like Bela more than BeLa! I'll correct it everywhere in this PR, but I only used BeLa because I've found it in other places...

About the multiple HelpFiles: I originally wanted to write extension files, but it doesn't work sense since we are not writing an extension. It could be an idea though to package Bela's UGens and sclang additions as an extension, that would help keeping all Bela code in one place, as opposed to intersperse it all over the place. 
What do you think? Let's talk about this before merge-time?

File: HelpSource/Classes/Array.schelp (hunk 1 above)
Comment (elgiano) [reply HelpSource/Classes/Array.schelp]: sure, sorry! thanks

Comment (giuliomoro) [reply HelpSource/Classes/Array.schelp]: @brianlheim does the linting tool not process these files? Should it?

**Comment by giuliomoro** [comment]
>  since we are not writing an extension. It could be an idea though to package Bela's UGens and sclang additions as an extension, that would help keeping all Bela code in one place, as opposed to intersperse it all over the place.
What do you think? Let's talk about this before merge-time?

That sounds interesting, how would it work? Would this be a separate repo which only contains some Ugens and class files? Would you be able to explain to me exactly what would go there and what would remain in the main `supercollider` repo?

File: HelpSource/Classes/Array.schelp (hunk 1 above)
Comment (mossheim) [reply HelpSource/Classes/Array.schelp]: the clang-format.py script only processes c++ files; clang-format doesn't know anything about schelp syntax.

there are a number of things that would be good to check in linting the entire codebase. right now it's just clang-format, though.

**Comment by mossheim** [comment]
> It could be an idea though to package Bela's UGens and sclang additions as an extension, that would help keeping all Bela code in one place, as opposed to intersperse it all over the place.
> What do you think? Let's talk about this before merge-time?

we should merge it into mainline. the new UGens are ones that anyone working on Bela would expect to have out of the box. plus, we don't have a good distribution model yet for UGen binaries. this would just make things harder for everyone.

**Comment by elgiano** [comment]
Ok! So, do I understand correctly that this PR is good as-is?

**Comment by mossheim** [comment]
i have no issue with it, but it's @giuliomoro 's call :)

**Comment by giuliomoro** [comment]
Yes thanks it looks good as it is. I will get around to merging all the outstanding ones early in the week probably.

**Pull request merged**